Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift

```diff
@@ -290,7 +290,7 @@ extension BrowserViewController: URLBarDelegate {
         } else {
             showSearchController()
         }
-
+        urlBar.locationTextField?.isPrivateMode = tabManager.selectedTab?.isPrivate ?? false
```

Comment:
Can we find a way to use instead the existing protocol `PrivateModeUI` which will update this `isPrivate` variable? That way we can keep the variable private and leverage what already exists.

Reply:
I thought about doing it that way in the beginning but we just needed that value to be set at init so it knows what color to use for the highlight, while `PrivateModeUI ` can't be called while the autocomplete label is shown because the user has no means of changing between private/normal mode until he dismisses the keyboard, in which case the autocomplete label is already off screen, which makes the protocol conformance a bit redundant for this case.